api/k8s/v1beta4/k8s_helpers.go

```diff
@@ -47,3 +47,62 @@ func (s *API) WaitForCluster(req *WaitForClusterRequest) (*Cluster, error) {
 	}
 	return cluster.(*Cluster), nil
 }
+
+// WaitForClusterNodesReadyRequest is used by WaitForClusterNodesReady method.
+type WaitForClusterNodesReadyRequest struct {
+	ClusterID string
+	Region    scw.Region
+	Timeout   time.Duration
+}
+
+// WaitForClusterNodesReady waits for the nodes of a cluster to be ready
+func (s *API) WaitForClusterNodesReady(req *WaitForClusterNodesReadyRequest) error {
+	pools, err := s.ListPools(&ListPoolsRequest{
+		ClusterID: req.ClusterID,
+		Region:    req.Region,
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, pool := range pools.Pools {
+		err = s.WaitForPoolNodesReady(&WaitForPoolNodesReadyRequest{
+			PoolID:  pool.ID,
+			Timeout: req.Timeout,
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// WaitForPoolNodesReadyRequest is used by WaitForPoolNodesReady method.
+type WaitForPoolNodesReadyRequest struct {
+	PoolID  string
+	Timeout time.Duration
+}
+
+// WaitForPoolNodesReady waits for the nodes of a pool to be ready
+func (s *API) WaitForPoolNodesReady(req *WaitForPoolNodesReadyRequest) error {
+	poolRequest := &GetPoolRequest{
+		PoolID: req.PoolID,
+	}
+
+	_, err := async.WaitSync(&async.WaitSyncConfig{
+		Get: func() (interface{}, bool, error) {
+			pool, err := s.GetPool(poolRequest)
+			if err != nil {
+				return nil, false, err
+			}
+
+			return nil, pool.Size == pool.CurrentNodeCount && pool.Status == "ready", nil
```

Comment:
No need to check the size and use the defined variable for the `"ready"` :)

Reply:
`PoolStatusReady`

Reply:
Updated